packages/schemas/src/types/system.ts

```diff
@@ -97,21 +97,50 @@ export const demoSocialGuard: Readonly<{
   [DemoSocialKey.DemoSocial]: demoSocialDataGuard,
 });
 
+// Cloudflare Hostnames
+export const customHostnamesDataGuard = z.object({
+  zoneId: z.string(),
+  apiToken: z.string(), // Requires zone permission for "SSL and Certificates Edit"
+});
+
+export type CloudflareHostnamesData = z.infer<typeof customHostnamesDataGuard>;
+
+export enum CloudflareKey {
+  CustomHostnames = 'cloudflareCustomHostnames',
+}
+
+export type CloudflareType = {
+  [CloudflareKey.CustomHostnames]: CloudflareHostnamesData;
```

Comment:
Should we give a better name? `CustomHostnames` can hardly be related to the data schema zoneId and API token?

Reply:
updated to "HostnameProvider"